addons/package_hierarchy/models/stock_quant_package.py

```diff
@@ -129,3 +132,19 @@ def action_toggle_processed(self):
             else:
                 for ml in move_lines:
                     ml.qty_done = 0
+
+    def action_view_picking(self):
+        """
+        Override method to be able to see package transfers from a parent package too.
+        Default filter will be confirmed picks:
+          state in ('confirmed', 'waiting', 'assigned')
+        """
+        action = self.env.ref("stock.action_picking_tree_all").read()[0]
+        packages = self.search([("id", "child_of", self.ids), ("package_id", "!=", False)])
+        domain = [
+            "|", ("result_package_id", "in", packages.ids), ("package_id", "in", packages.ids)
```

Comment:
Q: Do we need to consider `u_result_parent_package_id`? Might be one for product.

Reply:
When generally there is no issue result_package_id is a child of u_result_parent_package_id.
In the issues we have seen on multiple locations still is the case. Maybe when we were doing the changes in  a between time there was a moment where result_package_id was not a child of u_result_parent_package_id, which still after we fixed the final should be always.
Can add another or condition `("u_result_parent_package_id", "in", self.filtered(lambda p: not p.parent_id).ids)` -- Maybe better a check before if there are and to add with expressionor

Reply:
I think it's fine tbh.

It isn't possible now from the picking to have parent packages in two locations so if we had a parent package in two locations somehow then we would go to that record directly.